cpp/cmake_modules/DefineOptions.cmake

```diff
@@ -141,20 +141,22 @@ if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
 
   define_option(ARROW_BUILD_UTILITIES "Build Arrow commandline utilities" OFF)
 
-  define_option(ARROW_COMPUTE "Build the Arrow Compute Modules" ON)
+  define_option(ARROW_COMPUTE "Build the Arrow Compute Modules" OFF)
+
+  define_option(ARROW_CSV "Build the Arrow CSV Parser Module" OFF)
 
   define_option(ARROW_CUDA "Build the Arrow CUDA extensions (requires CUDA toolkit)" OFF)
 
-  define_option(ARROW_DATASET "Build the Arrow Dataset Modules" ON)
+  define_option(ARROW_DATASET "Build the Arrow Dataset Modules" OFF)
 
-  define_option(ARROW_FILESYSTEM "Build the Arrow Filesystem Layer" ON)
+  define_option(ARROW_FILESYSTEM "Build the Arrow Filesystem Layer" OFF)
 
   define_option(ARROW_FLIGHT
                 "Build the Arrow Flight RPC System (requires GRPC, Protocol Buffers)" OFF)
 
   define_option(ARROW_GANDIVA "Build the Gandiva libraries" OFF)
 
-  define_option(ARROW_HDFS "Build the Arrow HDFS bridge" ON)
+  define_option(ARROW_HDFS "Build the Arrow HDFS bridge" OFF)
```

Comment:
It seems that adding `-DARROW_HDFS=ON` explicitly is missing.

Reply:
I see, thanks, will fix